dubbo-remoting/dubbo-remoting-http12/src/main/java/org/apache/dubbo/remoting/http12/message/HttpMessageCodecFactory.java

```diff
@@ -27,12 +29,19 @@
 @SPI(scope = ExtensionScope.FRAMEWORK)
 public interface HttpMessageCodecFactory {
 
-    HttpMessageCodec createCodec(URL url, FrameworkModel frameworkModel);
+    HttpMessageCodec createCodec(URL url, FrameworkModel frameworkModel, HttpHeaders headers);
 
     MediaType contentType();
 
-    default boolean support(String contentType) {
+    default boolean supportDecode(HttpHeaders headers) {
+        String contentType = headers.getFirst(HttpHeaderNames.CONTENT_TYPE.getName());
         MediaType mediaType = this.contentType();
-        return mediaType.getName().startsWith(contentType);
+        return contentType.startsWith(mediaType.getName());
+    }
+
+    default boolean supportEncode(HttpHeaders headers) {
+        String acceptEncoding = headers.getFirst(HttpHeaderNames.ACCEPT.getName());
+        return acceptEncoding != null
+                && acceptEncoding.contains(this.contentType().getName());
```

Comment:
Maybe it shouldn't be removed `boolean support(String contentType)`, you can call `support(String)` with header value.